### PR TITLE
Fix incomplete description of callbacks.ModelCheckpoint

### PR DESCRIPTION
**Pull request by yil532**
Fix is based on this issue #41420
Two parts got fixed:

mode: "{auto, min, max}" should be {auto, min, max}
save_best_only: add the part that the current model will not be written

### CODE REVIEW

File: tensorflow/python/keras/callbacks.py [hunk 1]
```diff
@@ -1102,11 +1102,13 @@ class ModelCheckpoint(Callback):
         in the filename.
       monitor: quantity to monitor.
       verbose: verbosity mode, 0 or 1.
-      save_best_only: if `save_best_only=True`, the latest best model according
-        to the quantity monitored will not be overwritten.
-        If `filepath` doesn't contain formatting options like `{epoch}` then
-        `filepath` will be overwritten by each new better model.
-      mode: one of {auto, min, max}. If `save_best_only=True`, the decision to
+      save_best_only: if `save_best_only=True`, not only the "latest best model" 
+        according to the quantity monitored will not be overwritten, but also
```
Comment (rchao) [inline tensorflow/python/keras/callbacks.py:1106]: Thanks for the PR! I feel the updated description becomes a bit confusing. Can we say in the first sentence of `save_best_only` arg doc that it only saves when the model is considered the "best", and keep the original doc?

Comment (yil532) [reply tensorflow/python/keras/callbacks.py:1106]: Sure! Thank you for reviewing! How does this sound to you? @rchao 
  ```
save_best_only: if `save_best_only=True`, it only saves 
when the model is considered the "best" and the latest 
best model according to the quantity monitored will not 
be overwritten. If `filepath` doesn't contain formatting 
options like `{epoch}` then `filepath` will be overwritten 
by each new better model.
```

Comment (yil532) [reply tensorflow/python/keras/callbacks.py]: fixed!

File: tensorflow/python/keras/callbacks.py [hunk 2]
```diff
@@ -1102,11 +1102,12 @@ class ModelCheckpoint(Callback):
...
-        to the quantity monitored will not be overwritten.
-        If `filepath` doesn't contain formatting options like `{epoch}` then
-        `filepath` will be overwritten by each new better model.
-      mode: one of {auto, min, max}. If `save_best_only=True`, the decision to
+      save_best_only: if `save_best_only=True`, it only saves when the model 
+        is considered the "best" and the latest best model according to the 
+        quantity monitored will not be overwritten. If `filepath` doesn't contain 
+        formatting options like `{epoch}` then `filepath` will be overwritten 
+        by each new better model.
+      mode: one of `{auto, min, max}`. If `save_best_only=True`, the decision to
```
Comment (rchao) [inline tensorflow/python/keras/callbacks.py:1110]: Since auto, min, and max are string, I'd probably say {'auto', 'min', 'max'} here

**Review by rchao — DISMISSED** [review]
Looks good. Thanks!

File: tensorflow/python/keras/callbacks.py (hunk 2 above)
Comment (yil532) [reply tensorflow/python/keras/callbacks.py]: resolved! @rchao

**Review by rchao — DISMISSED** [review]
Looks good. Thanks for the PR!

**Comment by gbaned** [comment]
@yil532 Can you please address Ubuntu Sanity errors? Thanks!

**Review by omalleyt12 — APPROVED** [review]
Thanks for the PR!

**Pull request merged**